include/_kernel.h

```diff
@@ -32,6 +32,30 @@
 #endif
 #include "_kernel_c.h"
 
+/* If the -cl-std build option is not specified, the highest OpenCL C 1.x
+ * language version supported by each device is used as the version of
+ * OpenCL C when compiling the program for each device.
```

Comment:
I think it's OK to have different OpenCL version devices in the same platform. The devices do not have to match the host OpenCL version.